### ISSUE TITLE
addObject() throwing error: Path not supported by Algolia REST API. 

### ISSUE THREAD

**Issue by jimmyff** [knoxpo/dart_algolia#83]
I've just updated my project to support null safety, I'm unsure if that's relevant but since updating I'm getting this error when I attempt tp update an object in my algolia index:
```dart
          await algolia.index('myIndex').addObject(updatedObject);
```
Throws the following error:
```
I/flutter ( 3295): AlgoliaError(_message: {message: Path not supported by Algolia REST API. Please have a look at http://www.algolia.com/doc/rest_api for the list of valid commands, status: 404}, _statusCode: 404)
I/flutter ( 3295): #0      AlgoliaObjectReference.setData
package:algolia/src/object_reference.dart:62
I/flutter ( 3295): <asynchronous suspension>
I/flutter ( 3295): #1      AlgoliaIndexReference.addObject
package:algolia/src/index_reference.dart:150
I/flutter ( 3295): <asynchronous suspension>
I/flutter ( 3295): #2      AlgoliaRebuilder.patch
package:manage/…/algolia/algolia_rebuilder.dart:109
I/flutter ( 3295): <asynchronous suspension>
I/flutter ( 3295): #3      _DbSyncScreenState.executeStep2
package:manage/screens/db_sync_screen.dart:61
I/flutter ( 3295): <asynchronous suspension>
```

Currently using version 1.0.4 via flutter:
```
[✓] Flutter (Channel stable, 2.8.1, on Ubuntu 21.10
    5.13.0-27-generic, locale en_GB.UTF-8)
    • Flutter version 2.8.1 at /home/jimmyff/flutter
    • Upstream repository
      https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (7 weeks ago), 2021-12-16
      08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android
    SDK version 31.0.0)
    • Android SDK at /home/jimmyff/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at:
      /home/jimmyff/tools/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable
    at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a
      Chrome executable.

[✓] Android Studio (version 2020.3)
    • Android Studio at /home/jimmyff/tools/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      11.0.10+0-b96-7249189)

[✓] VS Code
    • VS Code at /snap/code/current
    • Flutter extension version 3.34.0

[✓] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 •
      android-x86 • Android 11 (API 30) (emulator)

```

**Comment by t-kietzmann** [comment]
Had a similar error on Web in release mode after upgrading from 1.0.2 to 1.0.4.

Use this in your pubspec.yaml
algolia: 1.0.2
and try again (force 1.0.2)

**Comment by nhathiwala** [comment]
This has been fix in `1.0.5`

**Comment by jimmyff** [comment]
@nhathiwala I'm getting this issue in `1.1.2` when using `addObject()`

![Screenshot 2024-02-26 at 11 56 35](https://github.com/knoxpo/dart_algolia/assets/298850/fa7dd53e-f1c5-4c20-b56e-550252f83049)

Any ideas how I can resolve it?
